src/Parser/HolyPriest/Modules/PriestCore/SerendipityReduction/SerenityReduction.js

```diff
@@ -0,0 +1,90 @@
+// dependencies
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS';
+import Module from 'Parser/Core/Module';
+
+// CHANGE THIS BETWEEN COPIES
+const HOLY_WORD_SPELL_ID = SPELLS.HOLY_WORD_SERENITY.id;
+
+// We are giving a buffer of 75% of CD due to the fact that the large
+// majority of players would intentionally use spells to push holy words
+// off cooldown if needed. This is a feelycraft metric that is open for
+// modification or outright removal depending on opinions.
+const FULL_OVERCAST_LENIENCE = 0.75;
+
+class SerenityReduction extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  }
+
+  // Holy Word reduction spells (aka things that apply the respective Serendipity)
+  serendipityProccers = {
+    [SPELLS.GREATER_HEAL.id]: 1.0,
+    [SPELLS.FLASH_HEAL.id]: 1.0,
+    [SPELLS.BINDING_HEAL_TALENT.id]: 0.5,
+  }
+
+  currentCooldown = 0
+  maxCooldown = 60000
+  serendipityReduction = 6000
+
+  holy_t20_2p = 0.0
+  holy_t20_2p_active = false;
+  overcast = 0.0 // Overall wasted serendipity
+  rawReduction = 0.0
+  casts = 0
+  effectiveFullOvercasts = 0
+  _tempOvercast = 0.0 // Tracker of how much wasted overcast between each holy word cast
+
+  on_initialized() {
+    // Set up proper serendipity reduction values
+    if (this.combatants.selected.hasTalent(SPELLS.LIGHT_OF_THE_NAARU_TALENT.id)) {
+      this.serendipityReduction += 2000;
+    }
+    if (this.combatants.selected.hasBuff(SPELLS.HOLY_PRIEST_T20_2SET_BONUS_BUFF.id)) {
+      this.serendipityReduction += 1000;
+      this.holy_t20_2p_active = true;
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId === HOLY_WORD_SPELL_ID) {
+      this.currentCooldown = event.timestamp + this.maxCooldown;
+      this.casts += 1;
+
+      // We are giving a buffer of 75% of CD due to the fact that the large
+      // majority of players would intentionally use spells to push holy words
+      // off cooldown if needed. This is a feelycraft metric that is open for
+      // modification or outright removal depending on opinions.
+      this.effectiveFullOvercasts += Math.floor(this._tempOvercast / 1000 / (this.maxCooldown * FULL_OVERCAST_LENIENCE));
+      this._tempOvercast = 0.0;
+    }
+
+    // const affectedSpellId = this.getAffectedSpell(spellId);
+    if (Object.keys(this.serendipityProccers).indexOf(spellId.toString()) !== -1) {
```

Comment:
Should be `if(spellId.toString() in this.serendipityProccers)`, or alternately you could do 
```
const serendipityReductionFactor = this.serendipityProccers[spellId];
if(serendipityReductionFactor) {
...
```

Reply:
ah yes, was too used to the structure I've been mimicking from other modules